ci-operator/step-registry/aks/provision/aks-provision-commands.sh

```diff
@@ -6,22 +6,29 @@ AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
 AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
 AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
 AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+AZURE_LOCATION="$LEASED_RESOURCE"
 
-CLUSTER="${NAMESPACE}-${UNIQUE_HASH}"
-RESOURCEGROUP="$(<"${SHARED_DIR}/resourcegroup")"
+RESOURCE_NAME_PREFIX="${NAMESPACE}-${UNIQUE_HASH}"
 
 az --version
 az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
 
+set -x
+
+echo "Creating resource group for the aks cluster"
+RESOURCEGROUP="${RESOURCE_NAME_PREFIX}-aks-rg"
+az group create --name "$RESOURCEGROUP" --location "$AZURE_LOCATION"
+
 echo "Building up the aks create command"
+CLUSTER="${RESOURCE_NAME_PREFIX}-aks-cluster"
 AKE_CREATE_COMMAND=(
     az aks create
     --name "$CLUSTER"
     --resource-group "$RESOURCEGROUP"
     --node-count "$AKS_NODE_COUNT"
     --load-balancer-sku "$AKS_LB_SKU"
     --os-sku "$AKS_OS_SKU"
-    --location "$LEASED_RESOURCE"
+    --location "$AZURE_LOCATION"
 )
```

Comment:
Any reason not to also enable FIPS?

Reply:
The enablement of FIPS is controlled by the `AKS_ENABLE_FIPS_IMAGE` environment variable, which defaults to true. 

This is set in lines 31-33 of the script:
```bash
if [[ "$AKS_ENABLE_FIPS_IMAGE" == "true" ]]; then
    AKE_CREATE_COMMAND+=(--enable-fips-image)
fi
```

Reply:
Sounds good. It's not enabled though when we create the HC right? Or am I missing that?

Reply:
Found this in the [rehearsal logs](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/53902/rehearse-53902-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-azure-aks-hypershift-fips-guest-f7/1808383774107373568/artifacts/azure-aks-hypershift-fips-guest-f7/aks-provision/build-log.txt):
```
++ az aks create --name ci-op-sfyzkzgf-e26bc-aks-cluster --resource-group ci-op-sfyzkzgf-e26bc-aks-rg --node-count 3 --load-balancer-sku standard --os-sku AzureLinux --location southcentralus --generate-ssh-keys --enable-fips-image
```

I think it was enabled in the rehearsal

Reply:
No I mean on the HC create cluster command

Reply:
On that command FIPS is enabled as well:
```
+ /usr/bin/hypershift create cluster azure --fips ...
```

See [rehearsal logs](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/53902/rehearse-53902-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-azure-aks-hypershift-fips-guest-f7/1808383774107373568/artifacts/azure-aks-hypershift-fips-guest-f7/create-hostedcluster/build-log.txt).

Reply:
The `--fips` flag is set here:
https://github.com/openshift/release/blob/a2003bca867eaafadf2fc33e6ddee45b41d99de0/ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.16__amd64-nightly.yaml#L1522

Reply:
Ah ok, sorry, I was missing this. 👍🏻